### PR TITLE
Unmasking frame on websocket

### PR DESCRIPTION
**Pull request by stecolet**
Signed-off-by: stecolet@gmail.com <stecolet@gmail.com>

### CODE REVIEW

**Comment by stecolet** [comment]
As you can see from https://tools.ietf.org/html/rfc6455#section-5.2 websocket protocol supports also unmasked frame while Vetx 3.3.0 not. In this PR I've configured Netty properly to support unmasked frame when unmaskedFrame flasg is enabled in httpServer/client options

**Comment by vietj** [comment]
rather name the option "performFrameMasking" or "performFrameUnmasking"

**Comment by stecolet** [comment]
As you can see from last commit I've renamed option to performUnmaskingFrame.

**Comment by vietj** [comment]
almost good : we need to have equals/hashcode in HttpClientOptions and HttpServerOptions and these tested in the options test in Http1xTest

**Comment by stecolet** [comment]
Yes, I'm very agree. I've done.

Thank you
Stefano

**Comment by vietj** [comment]
looking at the interest of the PR, I don't understand why you need this feature as the RFC6455 clearly says that the client frames must be masked and the server frames must not be masked.

**Comment by stecolet** [comment]
In my project there's a WebSocket communication between two server side services. Maybe this kind of protocol shoud not be the best way, but I can't change it because I'm going to integrate an existing service (which are using WebSocket). For this reason my need is to create a WebSocket server accepting unmasked frame.

Am I Clear??
Stefano.

**Comment by vietj** [comment]
yes it is clear now

**Comment by vietj** [comment]
having read the RFC more carefully, we should rename the options to have something more clear:

- client option should be called : `sendUnmaskedFrames` default = false
- server option should be called : `acceptUnmaskedFrames` default = false

There should be more documentation:

- the client documentation must say that this option for sending frames only and by default masked frames and activating this option bypasses the RFC
- the server documentation must say that this option is for accepting frames only and by default unmasked frames are rejected and activating this option bypasses the RFC

**Comment by stecolet** [comment]
@vietj, When you say "more documentation", it means to explain the behaviour only by JavaDoc onthe set method of the field? or more documentation on others doc file ?

**Comment by vietj** [comment]
@stecolet javadoc only, as this is really specific behavior for a non general use case, it's better to avoid the general reference guide

File: src/main/java/io/vertx/core/http/HttpClientOptions.java [hunk 1]
```diff
@@ -138,6 +138,11 @@
    */
   public static final boolean DEFAULT_HTTP2_CLEAR_TEXT_UPGRADE = true;
 
+  /**
+   * Default WebSocket Masked bit is true as depicted by RFC. PerformingUnMasking will be false
+   */
+  public static final boolean DEFAULT_FRAME_UNMASKING = false;
```
Comment (vietj) [inline src/main/java/io/vertx/core/http/HttpClientOptions.java]: last but not least rename `DEFAULT_FRAME_UNMASKING` to `DEFAULT_SEND_UNMASKED_FRAMES`

File: src/main/java/io/vertx/core/http/HttpServerOptions.java [hunk 2]
```diff
@@ -94,6 +94,11 @@
    */
   public static final boolean DEFAULT_DECOMPRESSION_SUPPORTED = false;
 
+  /**
+   * Default WebSocket Masked bit is true as depicted by RFC. PerformingUnMasking will be false
+   */
+  public static final boolean DEFAULT_FRAME_UNMASKING = false;
```
Comment (vietj) [inline src/main/java/io/vertx/core/http/HttpServerOptions.java]: last but not least rename `DEFAULT_FRAME_UNMASKING` to `DEFAULT_ACCEPT_UNMASKED_FRAMES`

Comment (stecolet) [inline src/main/java/io/vertx/core/http/HttpServerOptions.java]: Sure, I'm sorry, I forgot it.
Thank you for your support.
Stefano

**Comment by vietj** [comment]
thanks for the contribution

**Pull request merged**